test/CarouselSpec.js

```diff
@@ -315,6 +315,39 @@ describe('<Carousel>', () => {
       expect(onSelectSpy).to.have.been.calledOnce;
     });
 
+    it('should go through the items given the specified intervals', () => {
+      const defaultInterval = 5000;
+      const intervals = [
+        1000,
+        500,
+        defaultInterval, // to test for no interval specified, it should be the default
+      ];
+      const itemsWithIntervals = [
+        <Carousel.Item key={1} interval={intervals[0]}>
+          Item 1 content
+        </Carousel.Item>,
+        <Carousel.Item key={2} interval={intervals[1]}>
+          Item 2 content
+        </Carousel.Item>,
+        <Carousel.Item key={3}>Item 3 content</Carousel.Item>,
+      ];
+
+      const onSelectSpy = sinon.spy();
+      mount(
+        <Carousel interval={defaultInterval} onSelect={onSelectSpy}>
+          {itemsWithIntervals}
+        </Carousel>,
+      );
+
+      const total = intervals.reduce((sum, current) => sum + current, 0);
+      clock.tick(total * 1.1);
```

Comment:
Yep, this is still causing consistency issues 😆 anyone gotta better idea for a more consistent approach to testing this feature?

Reply:
not really... and this isn't directly asserting that item intervals individually get respected.

maybe just set something up where the carousel has an interval of `1000`, the first item has an interval of `100`, and make sure that we're on the 2nd item after `200` units have lapsed?